priv/rest.ex.eex

```diff
@@ -165,15 +157,19 @@ defmodule <%= context.module_name %> do
     "#{proto}://#{host}:#{port}#{path}"
   end
 
-  defp add_query(url, []) do
+  defp add_query(url, [], _client) do
     url
   end
-  defp add_query(url, query) do
-    querystring = AWS.Util.encode_query(query)
+  defp add_query(url, query, client) do
+    querystring = encode!(client, query, :query)
```

Comment:
I don't think we to change this one. Query encoding is something separate and I don't see a reason why people would want to customize it. :)

Reply:
There are some cases such as RDS which encodes the payload as form data. So given that we will have to cover "query" cases anyway, I thought it wouldn't hurt to delegate this call and reuse this.

Here is the RDS example:
https://github.com/aws-beam/aws-elixir/pull/42/files#diff-bbf2726032cab768e2602d9a603d52373019f1237ecaa633c1e29e73797487b7R2017
```elixir
  defp encode!(client, payload) do
    AWS.Client.encode!(client, payload, :query)
  end

  defp decode!(client, payload) do
    AWS.Client.decode!(client, payload, :xml)
  end
```

Reply:
What do you think? Should we keep this querystring part pointing to AWS.XML module?

Reply:
Since it is always query string encoding, which is relatively straight-forward, I would skip it. But we can also tackle this on on AWS.Client, so your call.

Reply:
Ok. I will keep it then. We can even get rid of AWS.Util in the future or revert this later, if anything.